libraries/botbuilder-ai/src/luis-client/luisClient.ts

```diff
@@ -0,0 +1,301 @@
+
+        this.authentications.default.applyToRequest(localVarRequestOptions);
+
+        if (Object.keys(localVarFormParams).length) {
+            if (localVarUseFormData) {
+                (<any>localVarRequestOptions).formData = localVarFormParams;
+            } else {
+                localVarRequestOptions.form = localVarFormParams;
+            }
+        }
+        return new Promise<LuisResult>((resolve, reject) => {
+            localVarRequest(localVarRequestOptions, (error, response, luisResult) => {
+                if (error) {
+                    reject(error);
+                } else {
+                    luisResult = ObjectSerializer.deserialize(luisResult, 'LuisResult');
+                    if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
+                        resolve(luisResult);
+                    } else {
+                        reject({
+                            response: {
+                                response: response,
+                                headers:
+                                    response.headers,
+                                body: response.body,
+                                status:
+                                    response.statusCode
+                            }
+                        });
+                    }
+                }
+            });
+        });
+    }
+    /**
+     * Gets predictions for a given utterance, in the form of intents and entities. The current maximum query size is 500 characters.
+     * @param appId The LUIS application ID (guid).
+     * @param query The utterance to predict.
+     * @param [options] The optional parameters
+     * @returns Promise<LuisResult>
+     */
+    public async predictionResolveGet(appId: string, query: string, options: PredictionResolveOptionalParams): Promise<LuisResult> {
+        const localVarPath = this.basePath + '/apps/{appId}'
+            .replace('{' + 'appId' + '}', encodeURIComponent(String(appId)));
+        let localVarQueryParameters: any = {};
+        let localVarHeaderParams = (Object as any).assign({}, this.defaultHeaders) as any;
+        let localVarFormParams: any = {};
+
+        // verify required parameter 'appId' is not null or undefined
+        if (appId === null || appId === undefined) {
+            throw new Error('Required parameter appId was null or undefined when calling predictionResolve2.');
+        }
+
+        // verify required parameter 'q' is not null or undefined
+        if (query === null || query === undefined) {
+            throw new Error('Required parameter q was null or undefined when calling predictionResolve2.');
+        }
+
+        if (options.customHeaders === null || options.customHeaders === undefined) {
+            options.customHeaders = {headers: {}}
+        }
+
+        if (query !== undefined) {
+            localVarQueryParameters['q'] = ObjectSerializer.serialize(query, 'string');
```

Comment:
Is it ok that if the parameter is `query` here it looks for `q`?

Reply:
Nice catch. We will review it. Thanks!